compiler/semantic/expr.go

```diff
@@ -213,13 +214,27 @@ func (a *analyzer) semExpr(e ast.Expr) dag.Expr {
 					Name:  elem.Name,
 					Value: v,
 				})
+				continue
 			case *ast.Spread:
 				e := a.semExpr(elem.Expr)
 				out = append(out, &dag.Spread{
 					Kind: "Spread",
 					Expr: e,
 				})
+				continue
 			}
+			e := a.semExpr(elem)
+			name := inferColumnName(e, elem)
+			if _, ok := fields[name]; ok {
+				a.error(elem, fmt.Errorf("record expression: %w", &super.DuplicateFieldError{Name: name}))
+				continue
+			}
+			fields[name] = struct{}{}
+			out = append(out, &dag.Field{
+				Kind:  "Field",
+				Name:  name,
+				Value: e,
+			})
```

Comment:
Nit: I think what's going on would be a little clearer if you promote this block to be the default case in the preceding type switch instead of adding continue at the end of its existing cases.